src/linear_algebra/multilinear.lean

```diff
@@ -135,6 +145,63 @@ variables [comm_ring R] [∀i, add_comm_group (M₁ i)] [∀i, add_comm_group (M
 [∀i, module R (M i)] [∀i, module R (M₁ i)] [module R M₂]
 (f f' : multilinear_map R M₁ M₂)
 
+lemma map_piecewise_smul (c : ι → R) (m : Πi, M₁ i) (s : finset ι) :
+  f (s.piecewise (λi, c i • m i) m) = s.prod c • f m :=
+begin
+  refine s.induction_on (by simp) _,
+  assume j s j_not_mem_s Hrec,
+  have A : function.update (s.piecewise (λi, c i • m i) m) j (m j) =
+           s.piecewise (λi, c i • m i) m,
+  { ext i,
+    by_cases h : i = j,
+    { rw h, simp [j_not_mem_s] },
+    { simp [h] } },
+  rw [s.piecewise_insert, f.map_smul, A, Hrec],
+  simp [j_not_mem_s, mul_smul]
+end
+
+/-- Multiplicativity of a multilinear map along all coordinates at the same time,
+writing `f (λi, c i • m i)` as `univ.prod c • f m`. -/
+lemma map_smul_univ [fintype ι] (c : ι → R) (m : Πi, M₁ i) :
+  f (λi, c i • m i) = finset.univ.prod c • f m :=
+by simpa using map_piecewise_smul f c m finset.univ
+
+lemma map_piecewise_add (m m' : Πi, M₁ i) (t : finset ι) :
```

Comment:
It took me some time to parse RHS. It would be nice to have some comment explaining the meaning of this lemma.

Reply:
Done